operators/pkg/instance-controller/controller.go

```diff
@@ -156,12 +161,62 @@ func (r *InstanceReconciler) setInstanceStatus(
 	klog.Info(msg)
 	r.EventsRecorder.Event(instance, eventType, eventReason, msg)
 
-	instance.Status.Phase = eventReason
-	instance.Status.IP = ip
-	instance.Status.URL = url
-	instance.Status.ObservedGeneration = instance.ObjectMeta.Generation
-	if err := r.Status().Update(ctx, instance); err != nil {
+	type InstanceStatusSpec struct {
+		Op    string                           `json:"op"`
+		Path  string                           `json:"path"`
+		Value crownlabsv1alpha2.InstanceStatus `json:"value,omitempty"`
+	}
+
+	status := make([]InstanceStatusSpec, 1)
+	status[0].Op = "replace"
+	status[0].Path = "/status"
+	status[0].Value = crownlabsv1alpha2.InstanceStatus{
+		IP:                 ip,
+		Phase:              eventReason,
+		URL:                url,
+		ObservedGeneration: instance.ObjectMeta.Generation,
+	}
+
+	patchBytes, _ := json.Marshal(status)
+
+	if err := r.Status().Patch(ctx, instance, client.RawPatch(types.JSONPatchType, patchBytes)); err != nil {
 		klog.Error("Unable to update Instance status")
 		klog.Error(err)
 	}
 }
+
+// UpdateInstanceLabel update instance labels.
+func (r *InstanceReconciler) UpdateInstanceLabel(ctx context.Context, template *crownlabsv1alpha2.Template, instance *crownlabsv1alpha2.Instance) error {
+	type instanceLabSpec struct {
+		Op    string            `json:"op"`
+		Path  string            `json:"path"`
+		Value map[string]string `json:"value,omitempty"`
+	}
+
+	instanceLab := make([]instanceLabSpec, 1)
+	instanceLab[0].Op = "replace"
+	instanceLab[0].Path = "/metadata/labels"
+	instanceLab[0].Value = map[string]string{
+		"course-name":        strings.ReplaceAll(strings.ToLower(template.Spec.WorkspaceRef.Name), " ", "-"),
+		"template-name":      template.Name,
+		"template-namespace": template.Namespace,
```

Comment:
I would propose the following labels, to better align with the new names:
* `crownlabs.polito.it/workspace`: template.Spec.WorkspaceRef.Name (the replace should no longer be needed, as a valid resource name is also a valid label value)
* `crownlabs.polito.it/template`: template.Name (the workspace is already implied by the workspace)
* `crownlabs.polito.it/managed-by`: `instance`